code/game/objects/items/props/helmetgarb.dm

```diff
@@ -94,110 +116,351 @@
 
 	remove_attached_item()
 
-	attached_item = S.master_object
+	var/obj/item/MO = S.master_object
+	if(!istype(MO, /obj/item/clothing/head/helmet/marine) && !istype(MO, /obj/item/clothing/head/cmcap)) // Do not bother if it's not a helmet or at least a hat
+		return
+
+	attached_item = MO
+
 	RegisterSignal(attached_item, COMSIG_PARENT_QDELETING, .proc/remove_attached_item)
-	activation = new /datum/action/item_action/toggle(src, S.master_object)
+	RegisterSignal(attached_item, COMSIG_ITEM_EQUIPPED, .proc/toggle_check)
 
-	if(ismob(S.master_object.loc))
-		activation.give_to(S.master_object.loc)
+	if(ismob(attached_item.loc))
+		set_attached_mob(attached_item.loc)
 
-/obj/item/prop/helmetgarb/helmet_nvg/on_exit_storage(obj/item/storage/S)
-	remove_attached_item()
-	return ..()
 
-/obj/item/prop/helmetgarb/helmet_nvg/proc/remove_attached_item()
-	SIGNAL_HANDLER
-	if(!attached_item)
-		return
+/obj/item/prop/helmetgarb/helmet_nvg/attackby(var/obj/item/A as obj, mob/user as mob)
+	if(istype(A,/obj/item/cell))
+		recharge()
 
-	UnregisterSignal(attached_item, COMSIG_PARENT_QDELETING)
-	qdel(activation)
-	attached_item = null
+	if(HAS_TRAIT(A, TRAIT_TOOL_SCREWDRIVER))
+		repair()
 
-/obj/item/prop/helmetgarb/helmet_nvg/ui_action_click(var/mob/owner, var/obj/item/holder)
-	toggle_nods(owner)
-	activation.update_button_icon()
+	else
+		..()
 
+/obj/item/prop/helmetgarb/helmet_nvg/proc/recharge(var/obj/item/A as obj, mob/user as mob)
+	if(user.action_busy)
+		return
+	if(src != user.get_inactive_hand())
+		to_chat(user, SPAN_WARNING("You need to hold \the [src] in hand in order to recharge them."))
+		return
+	if(shape == NVG_SHAPE_COSMETIC)
+		to_chat(user, SPAN_WARNING("There is no connector for the power cell inside \the [src]."))
+		return
+	if(shape == NVG_SHAPE_BROKEN)
+		to_chat(user, SPAN_WARNING("You need to repair \the [src] first."))
+		return
+	if(nvg_charge == nvg_maxcharge)
+		to_chat(user, SPAN_WARNING("\The [src] are already fully charged."))
+		return
 
-/obj/item/prop/helmetgarb/helmet_nvg/proc/toggle_nods(var/mob/user)
-	if(user.is_mob_incapacitated())
+	var/obj/item/cell/C = A
+
+	while(nvg_charge < nvg_maxcharge)
+		if(C.charge <= 0)
+			to_chat(user, SPAN_WARNING("\The [C] is completely dry."))
+			break
+		var/to_transfer
+		if(do_after(user, 1 SECONDS, (INTERRUPT_ALL & (~INTERRUPT_MOVED)), BUSY_ICON_BUILD, C, INTERRUPT_DIFF_LOC))
+			to_transfer = min(400, C.charge, (nvg_maxcharge - nvg_charge))
+			C.charge -= to_transfer
+			nvg_charge += to_transfer
+			to_chat(user, "You transfer some power between \the [C] and \the [src]. The gauge now reads: [round(100.0*nvg_charge/nvg_maxcharge) ]%.")
+		else
+			to_chat(user, SPAN_WARNING("You were interrupted."))
+			break
+
+/obj/item/prop/helmetgarb/helmet_nvg/proc/repair(var/obj/item/A as obj, mob/user as mob)
```

Comment:
```suggestion
/obj/item/prop/helmetgarb/helmet_nvg/proc/repair(mob/user)
```
The item being used for repairs is not relevant to anything in the procedure, if I am reading it correctly?

Reply:
correct the screwdriver check was made before . i applied this suggestion

Reply:
correct the screwdriver check was made before . i applied this suggestion